server/analyzer/resolve_type.go

```diff
@@ -70,6 +91,37 @@ func ResolveType(ctx *sql.Context, a *analyzer.Analyzer, node sql.Node, scope *p
 	})
 }
 
+// ResolveTypeForExprs replaces types.ResolvableType to appropriate types.DoltgresType.
+func ResolveTypeForExprs(ctx *sql.Context, a *analyzer.Analyzer, node sql.Node, scope *plan.Scope, selector analyzer.RuleSelector, qFlags *sql.QueryFlags) (sql.Node, transform.TreeIdentity, error) {
+	return pgtransform.NodeExprsWithOpaque(node, func(expr sql.Expression) (sql.Expression, transform.TreeIdentity, error) {
+		var same = transform.SameTree
+		switch e := expr.(type) {
+		case *expression.ExplicitCast:
```

Comment:
We should handle Assignment and Implicit right? Should probably make TODOs for those too